### PR TITLE
feat: testnet mode

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes BX-1076
Figma link (if any):

https://www.figma.com/file/hzfwKOddfRmeVO4KUsG9wM/branch/GDDUZ9kbFUvmcFyvzMQNOl/Rainbow-BX?node-id=9193%3A279613&mode=dev

## What changed (plus any additional context for devs)

Testnet mode

- green boders and top testnet bar all over the extension except for unlock screen
- networks filtering depending on testnet mode and networks selected in settings
- transactions filtering depending on testnet mode and networks selected in settings
- assets filtering depending on testnet mode and networks selected in settings
- testnet mode shortcut enabled or disabled depending on settings
- dapp requests on testnet mode, green borders + top bar should send txs to testnets

## Screen recordings / screenshots

https://www.loom.com/share/377edd32106d45fb99a08bbe8710bfb3

### CODE REVIEW

File: src/entries/popup/components/TestnetBar/TestnetBar.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                style={{
+                  overflow: 'clip',
+                }}
+                marginRight="-12px"
+              >
+                <TestnetBarBackground />
+              </Box>
+              <MenuItem.TextIcon icon="🕹" />
+              <Text align="center" color="green" size="12pt" weight="heavy">
+                Testnet Mode
```
Comment (derHowie) [inline src/entries/popup/components/TestnetBar/TestnetBar.tsx:59]: any sense in putting this in i18n?

File: src/entries/popup/pages/home/TabHeader.tsx [hunk 2]
```diff
@@ -86,7 +86,7 @@ export function TabHeader({
         display="flex"
         justifyContent="space-between"
         paddingHorizontal="20px"
-        style={{ maxHeight: 11, textTransform: 'capitalize' }}
+        // style={{ maxHeight: 11, textTransform: 'capitalize' }}
```
Comment (derHowie) [inline src/entries/popup/pages/home/TabHeader.tsx:89]: 👀 was this maxHeight breaking UI when the testnet bar is showing? Would prefer to keep it if we can.

File: static/json/languages/en_US.json [hunk 3]
```diff
@@ -178,7 +186,9 @@
           "warning_3": "Anyone with your Recovery Phrase can access your entire wallet.",
           "warning_4": "Rainbow Support will never ask you for your Recovery Phrase.",
           "show": "Show Recovery Phrase",
-          "phrase_copied": "Recovery Phrase Copied"
+          "phrase_copied": "Recovery Phrase Copied",
+          "copied": "Recovery Phrase Copied"
```
Comment (derHowie) [inline static/json/languages/en_US.json:190]: this looks like a dupe

**Review by derHowie — APPROVED** [review]
You're a beast

File: src/entries/popup/components/TestnetBar/TestnetBar.tsx (hunk 1 above)
Comment (estebanmino) [reply src/entries/popup/components/TestnetBar/TestnetBar.tsx:59]: changing now

File: src/entries/popup/pages/home/TabHeader.tsx (hunk 2 above)
Comment (estebanmino) [reply src/entries/popup/pages/home/TabHeader.tsx:89]: rolling back

Comment (estebanmino) [reply src/entries/popup/pages/home/TabHeader.tsx:89]: oh ok rolling back

File: static/json/languages/en_US.json (hunk 3 above)
Comment (estebanmino) [reply static/json/languages/en_US.json:190]: right

File: src/entries/popup/pages/send/ToAddressInput.tsx [hunk 4]
```diff
@@ -358,7 +361,7 @@ export const ToAddressInput = React.forwardRef<InputRefAPI, ToAddressProps>(
       <>
         <DropdownInputWrapper
           zIndex={2}
-          dropdownHeight={452}
+          dropdownHeight={452 - (testnetMode ? TESTNET_MODE_BAR_HEIGHT : 0)}
```
Comment (brunobar79) [inline src/entries/popup/pages/send/ToAddressInput.tsx:364]: Should we move this 452 into core utils dimensions? Not sure if we're planning to use anywhere else

**Review by brunobar79 — APPROVED** [review]
Beast! 💯

File: src/entries/popup/pages/send/ToAddressInput.tsx (hunk 4 above)
Comment (estebanmino) [reply src/entries/popup/pages/send/ToAddressInput.tsx:364]: hm i don't think we have the same height anywhere else

**Pull request merged**